diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-gumstix.dts b/arch/arm/boot/dts/imx6dqscm-1gb-gumstix.dts
index 24fe0c6..b6d8142 100644
--- a/arch/arm/boot/dts/imx6dqscm-1gb-gumstix.dts
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-gumstix.dts
@@ -172,6 +172,30 @@
             <&gpio4 17 GPIO_ACTIVE_LOW>;
         compatible = "mmc-pwrseq-simple";
     };
+
+	v4l2_cap_0 {
+		compatible = "fsl,imx6q-v4l2-capture";
+		ipu_id = <0>;
+		csi_id = <0>;
+		mclk_source = <0>;
+		status = "okay";
+	};
+
+	v4l2_cap_1 {
+		compatible = "fsl,imx6q-v4l2-capture";
+		ipu_id = <0>;
+		csi_id = <1>;
+		mclk_source = <0>;
+		status = "okay";
+	};
+};
+
+&mipi_csi {
+	status = "okay";
+	ipu_id = <0>;
+	csi_id = <1>;
+	v_channel = <0>;
+	lanes = <2>;
 };
 
 &usbphy1 {
@@ -517,6 +541,21 @@
     pinctrl-names = "default";
     pinctrl-0 = <&i2c3_default>;
     clock-frequency = <400000>;
+
+	ov564x_mipi: ov564x_mipi@3c {
+		compatible = "ovti,ov564x_mipi";
+		reg = <0x3c>;
+		clocks = <&clks 201>;
+		clock-names = "csi_mclk";
+		pwn-gpios = <&gpio1 29 1>;   /* active low */
+		rst-gpios = <&gpio1 30 0>;   /* active high */
+		csi_id = <1>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&ov5640_default>;
+		status = "okay";
+	};
 };
 
 // ethernet device
@@ -737,6 +776,7 @@
         i2c3_default: i2c3grp {
             fsl,pins = <
                 MX6QDL_PAD_GPIO_5__I2C3_SCL PIN_OUTPUT_PULLUP /* gpio_5.i2c3_scl */
+                MX6QDL_PAD_GPIO_6__I2C3_SDA PIN_OUTPUT_PULLUP /* gpio_6.i2c3_sda */
             >;
         };
 
@@ -786,5 +826,13 @@
                 MX6QDL_PAD_DI0_DISP_CLK__GPIO4_IO16 PIN_INPUT /* di0_disp_clk.gpio4_io16 */
             >;
         };
+
+        ov5640_default: ov5640grp {
+            fsl,pins = <
+        		MX6QDL_PAD_ENET_TXD0__GPIO1_IO30 PIN_OUTPUT_PULLUP
+        		MX6QDL_PAD_ENET_TXD1__GPIO1_IO29 PIN_OUTPUT_PULLUP
+        		MX6QDL_PAD_GPIO_3__XTALOSC_REF_CLK_24M PIN_OUTPUT
+            >;
+        };
     };
 };
