From 76a6b804edc54cf78f4a6250e79108445ff61b82 Mon Sep 17 00:00:00 2001
From: ayaka <ayaka@soulik.info>
Date: Sun, 7 Sep 2014 04:40:36 +0800
Subject: [PATCH 01/12] v4l2: modify GstV4l2VideoDecCData to GstV4l2VideoCData

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 sys/v4l2/gstv4l2videodec.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sys/v4l2/gstv4l2videodec.c b/sys/v4l2/gstv4l2videodec.c
index d148b66..bc2682d 100644
--- a/sys/v4l2/gstv4l2videodec.c
+++ b/sys/v4l2/gstv4l2videodec.c
@@ -45,7 +45,7 @@ typedef struct
   gchar *device;
   GstCaps *sink_caps;
   GstCaps *src_caps;
-} GstV4l2VideoDecCData;
+} GstV4l2VideoCData;
 
 enum
 {
@@ -972,7 +972,7 @@ gst_v4l2_video_dec_subclass_init (gpointer g_class, gpointer data)
 {
   GstV4l2VideoDecClass *klass = GST_V4L2_VIDEO_DEC_CLASS (g_class);
   GstElementClass *element_class = GST_ELEMENT_CLASS (g_class);
-  GstV4l2VideoDecCData *cdata = data;
+  GstV4l2VideoCData *cdata = data;
 
   klass->default_device = cdata->device;
 
@@ -1008,9 +1008,9 @@ gst_v4l2_video_dec_register (GstPlugin * plugin, const gchar * basename,
   GTypeInfo type_info = { 0, };
   GType type, subtype;
   gchar *type_name;
-  GstV4l2VideoDecCData *cdata;
+  GstV4l2VideoCData *cdata;
 
-  cdata = g_new0 (GstV4l2VideoDecCData, 1);
+  cdata = g_new0 (GstV4l2VideoCData, 1);
   cdata->device = g_strdup (device_path);
   cdata->sink_caps = gst_caps_ref (sink_caps);
   cdata->src_caps = gst_caps_ref (src_caps);
-- 
2.7.4

