From b6e7bf1941bdb1a8712238c28f92ea1e01f9c866 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Tue, 3 Dec 2013 17:54:37 +0100
Subject: [PATCH 09/10] Pepper may still pass ATAGS to the kernel.

---
 include/configs/pepper.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/pepper.h b/include/configs/pepper.h
index 5a6a6c7..12123e4 100644
--- a/include/configs/pepper.h
+++ b/include/configs/pepper.h
@@ -13,6 +13,7 @@
 
 #include <asm/arch/omap.h>
 
+#define CONFIG_SETUP_MEMORY_TAGS 1
 /* Mach type */
 #define MACH_TYPE_PEPPER		4207	/* Until the next sync */
 #define CONFIG_MACH_TYPE		MACH_TYPE_PEPPER
-- 
1.7.10.4

