From 3012a33e234cd3e73961bff072424dd2ab480847 Mon Sep 17 00:00:00 2001
From: Yoruio <zeleidu@gmail.com>
Date: Tue, 11 Aug 2020 09:33:06 -0700
Subject: [PATCH] fixing compiler string checking by removing error translation

---
 src/core/fr-archive.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/fr-archive.c b/src/core/fr-archive.c
index 75f2309..7d48d39 100644
--- a/src/core/fr-archive.c
+++ b/src/core/fr-archive.c
@@ -1244,7 +1244,7 @@ copy_remote_file (FrArchive  *archive,
 	if (! g_file_query_exists (archive->file, archive->priv->cancellable)) {
 		GError *error;
 
-        error = g_error_new (G_IO_ERROR, G_IO_ERROR_NOT_FOUND, _("Archive not found"));
+        error = g_error_new (G_IO_ERROR, G_IO_ERROR_NOT_FOUND, "Archive not found");
 		fr_archive_copy_done (archive, FR_ACTION_LOADING_ARCHIVE, error);
 		g_error_free (error);
 
-- 
2.17.1

