From 5d0a7d621c40930e2ed3a2d39bf9f489dd73a9bf Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Tue, 6 May 2014 15:03:36 -0700
Subject: [PATCH 5/6] [HACK] Invert the y-axis of the touchscreen for pepper

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 drivers/input/touchscreen/ti_am335x_tsc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/touchscreen/ti_am335x_tsc.c b/drivers/input/touchscreen/ti_am335x_tsc.c
index 4e793a1..c2876c2 100644
--- a/drivers/input/touchscreen/ti_am335x_tsc.c
+++ b/drivers/input/touchscreen/ti_am335x_tsc.c
@@ -235,6 +235,7 @@ static void titsc_read_coordinates(struct titsc *ts_dev,
 			prev_val_x = read;
 
 		} else if (channel < creads * 2) {
+			read = 0xe93 - read + 0xc8;
 			diff = abs(read - prev_val_y);
 			if (diff < prev_diff_y) {
 				prev_diff_y = diff;
-- 
1.8.3.2

