From 6a70c112257d7c83d19dee5e1e39607441d020c7 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Tue, 6 May 2014 15:03:36 -0700
Subject: [PATCH 33/33] Invert the y-axis of the touchscreen for pepper

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 drivers/input/touchscreen/ti_am335x_tsc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/touchscreen/ti_am335x_tsc.c b/drivers/input/touchscreen/ti_am335x_tsc.c
index 2ce649520fe0..1f9812273ee7 100644
--- a/drivers/input/touchscreen/ti_am335x_tsc.c
+++ b/drivers/input/touchscreen/ti_am335x_tsc.c
@@ -235,6 +235,7 @@ static void titsc_read_coordinates(struct titsc *ts_dev,
 			prev_val_x = read;
 
 		} else if (channel < creads * 2) {
+			read = 0xe93 - read + 0xc8;
 			diff = abs(read - prev_val_y);
 			if (diff < prev_diff_y) {
 				prev_diff_y = diff;
-- 
2.1.4

