From 02a169d2927e3c02f27eb0c6bf83ee97cf809417 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Thu, 26 Jul 2012 14:27:16 +0200
Subject: [PATCH 26/30] board-overo: Add Caspa camera link frequencies

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 arch/arm/mach-omap2/board-overo.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/mach-omap2/board-overo.c b/arch/arm/mach-omap2/board-overo.c
index 39a0ecd..055e59e 100644
--- a/arch/arm/mach-omap2/board-overo.c
+++ b/arch/arm/mach-omap2/board-overo.c
@@ -89,9 +89,18 @@ static void mt9v032_set_clock(struct v4l2_subdev *subdev, unsigned int rate)
 	isp->platform_cb.set_xclk(isp, rate, ISP_XCLK_A);
 }
 
+static const s64 mt9v032_link_freqs[] = {
+	13000000,
+	26600000,
+	27000000,
+	0,
+};
+
 static struct mt9v032_platform_data mt9v032_platform_data = {
 	.clk_pol = 0,
 	.set_clock = mt9v032_set_clock,
+	.link_freqs = mt9v032_link_freqs,
+	.link_def_freq = 26600000,
 };
 
 static struct i2c_board_info mt9v032_i2c_device = {
-- 
1.7.9.5

