Explicitly link the atomic library

Signed-off-by: Sicong Jia <jiasicong@gmail.com>
===================================================================

diff --git a/mesa-17.1.7/configure.ac b/mesa-17.1.7/configure.ac
index c5b3faa..cc1e4b6 100644
--- mesa-17.1.7/configure.ac
+++ mesa-17.1.7/configure.ac
@@ -801,7 +801,7 @@ AC_CHECK_FUNC([dlopen], [DEFINES="$DEFINES -DHAVE_DLOPEN"],
 AC_SUBST([DLOPEN_LIBS])
 
 dnl Check if that library also has dladdr
-save_LIBS="$LIBS"
+save_LIBS="$LIBS -latomic"
 LIBS="$LIBS $DLOPEN_LIBS"
 AC_CHECK_FUNCS([dladdr])
 LIBS="$save_LIBS"
 
