From 124cf5f2035ff70cd45b0b0f82a0f7f4a86f5f72 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Fri, 23 Jan 2015 16:40:15 -0800
Subject: [PATCH 3/7] omap3: overo: Add DTS support for TobiDuo expansion

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 board/overo/overo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/overo/overo.c b/board/overo/overo.c
index 0ba3ad7..c5f5724 100644
--- a/board/overo/overo.c
+++ b/board/overo/overo.c
@@ -283,6 +283,7 @@ int misc_init_r(void)
 		/* second lan chip */
 		enable_gpmc_cs_config(gpmc_lan_config, &gpmc_cfg->cs[4],
 		    0x2B000000, GPMC_SIZE_16M);
+		setenv("expansionname", "tobiduo");
 		break;
 	case GUMSTIX_PALO35:
 		printf("Recognized Palo35 expansion board (rev %d %s)\n",
-- 
2.1.0

