From b3ffc33d9bac2c55ee3df16c691d11966e6d33c0 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Mon, 19 Aug 2013 13:02:21 -0700
Subject: [PATCH 6/7] Correct typo in comment for NAND_BLOCK_SIZE

Upstream-Status: Pending

---
 include/configs/omap3_overo.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/omap3_overo.h b/include/configs/omap3_overo.h
index 3224d44..231b429 100644
--- a/include/configs/omap3_overo.h
+++ b/include/configs/omap3_overo.h
@@ -127,7 +127,7 @@
  *  xloader              4 * NAND_BLOCK_SIZE = 512 KiB
  *  uboot               14 * NAND_BLOCK_SIZE = 1792 KiB
  *  uboot environtment   2 * NAND_BLOCK_SIZE = 256 KiB
- *  linux               64 * NAND_BLOCK_SIE = 8 MiB
+ *  linux               64 * NAND_BLOCK_SIZE = 8 MiB
  *  rootfs              remainder
  */
 #define MTDIDS_DEFAULT "nand0=omap2-nand.0"
-- 
1.7.9.5

