From caeb2230d84a420dc70c2c30b4611f85c605d07f Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Fri, 6 Sep 2013 16:51:51 -0700
Subject: [PATCH 09/10] Change NAND offsets for 8MB kernel partition


Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 include/configs/omap3_overo.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/configs/omap3_overo.h b/include/configs/omap3_overo.h
index 231b429..ad9d2f1 100644
--- a/include/configs/omap3_overo.h
+++ b/include/configs/omap3_overo.h
@@ -172,8 +172,8 @@
 /* nand device jffs2 lives on */
 #define CONFIG_JFFS2_DEV		"nand0"
 /* start of jffs2 partition */
-#define CONFIG_JFFS2_PART_OFFSET	0x680000
-#define CONFIG_JFFS2_PART_SIZE		0xf980000	/* size of jffs2 */
+#define CONFIG_JFFS2_PART_OFFSET	0xA80000
+#define CONFIG_JFFS2_PART_SIZE		0xf580000	/* size of jffs2 */
 							/* partition */
 
 /* Environment information */
@@ -221,7 +221,7 @@
 		"bootm ${loadaddr}\0" \
 	"nandboot=echo Booting from nand ...; " \
 		"run nandargs; " \
-		"nand read ${loadaddr} 280000 400000; " \
+		"nand read ${loadaddr} 280000 800000; " \
 		"bootm ${loadaddr}\0" \
 
 #define CONFIG_BOOTCOMMAND \
-- 
1.7.10.4

