From 267e6f4ccb6f7b9cc48828b5184f7c887f9d0383 Mon Sep 17 00:00:00 2001
From: Arun Bharadwaj <arun@gumstix.com>
Date: Thu, 16 Apr 2015 11:14:29 -0700
Subject: [PATCH 05/10] omap3: overo: Add DTS support for TobiDuo expansion

Signed-off-by: Arun Bharadwaj <arun@gumstix.com>
Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 board/overo/overo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/overo/overo.c b/board/overo/overo.c
index 2845e7c..3f14b03 100644
--- a/board/overo/overo.c
+++ b/board/overo/overo.c
@@ -185,6 +185,7 @@ int misc_init_r(void)
 			expansion_config.revision,
 			expansion_config.fab_revision);
 		MUX_GUMSTIX();
+		setenv("expansionname", "tobiduo");
 		break;
 	case GUMSTIX_PALO35:
 		printf("Recognized Palo35 expansion board (rev %d %s)\n",
-- 
2.1.4

