From 1fbdd82af8568e601f6ef2647dbbfcc48d874ca0 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Wed, 15 Apr 2015 11:50:37 -0700
Subject: [PATCH 09/10] overo: Add support for Palo35 expansion board

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 board/overo/overo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/overo/overo.c b/board/overo/overo.c
index 3f14b03..d5ac1ad 100644
--- a/board/overo/overo.c
+++ b/board/overo/overo.c
@@ -193,6 +193,7 @@ int misc_init_r(void)
 			expansion_config.fab_revision);
 		MUX_GUMSTIX();
 		setenv("defaultdisplay", "lcd35");
+		setenv("expansionname", "palo35");
 		break;
 	case GUMSTIX_PALO43:
 		printf("Recognized Palo43 expansion board (rev %d %s)\n",
-- 
2.1.4

