From e0eeabf2cc8bbdd496deba9ac07374085ee9c0a6 Mon Sep 17 00:00:00 2001
From: Sicong Jia <sicong.jia@gumstix.com>
Date: Fri, 9 Feb 2018 16:28:20 -0800
Subject: [PATCH] Add pepper, duovero garret50c, poblano dtb files

Signed-off-by: Sicong Jia <sicong.jia@gumstix.com>
---
 arch/arm/boot/dts/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index c75bb57..93d98ec 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -624,6 +624,9 @@ dtb-$(CONFIG_SOC_AM33XX) += \
 	am335x-lxm.dtb \
 	am335x-nano.dtb \
 	am335x-pepper.dtb \
+	am335x-pepper-43c.dtb \
+	am335x-pepper-43r.dtb \
+	am335x-pepper-dvi.dtb \
 	am335x-phycore-rdk.dtb \
 	am335x-shc.dtb \
 	am335x-sbc-t335.dtb \
@@ -633,6 +636,7 @@ dtb-$(CONFIG_ARCH_OMAP4) += \
 	omap4-droid4-xt894.dtb \
 	omap4-duovero-parlor.dtb \
 	omap4-kc1.dtb \
+	omap4-duovero-garret50c.dtb \
 	omap4-panda.dtb \
 	omap4-panda-a4.dtb \
 	omap4-panda-es.dtb \
@@ -646,6 +650,7 @@ dtb-$(CONFIG_SOC_AM43XX) += \
 	am437x-gp-evm.dtb \
 	am437x-idk-evm.dtb \
 	am437x-sbc-t43.dtb \
+	am437x-poblano.dtb \
 	am437x-sk-evm.dtb
 dtb-$(CONFIG_SOC_OMAP5) += \
 	omap5-cm-t54.dtb \
-- 
2.7.4

