From a7f1324e1a5908e2ef5a80c2ead0755567887527 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Wed, 26 Aug 2015 14:26:41 -0700
Subject: [PATCH 4/6] overo: Ensure ethernet is initialized for Arbor43C

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 board/overo/overo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/overo/overo.c b/board/overo/overo.c
index 34bf265..04207c9 100644
--- a/board/overo/overo.c
+++ b/board/overo/overo.c
@@ -358,6 +358,7 @@ int board_eth_init(bd_t *bis)
 	case GUMSTIX_TOBI:
 	case GUMSTIX_CHESTNUT43:
 	case GUMSTIX_STAGECOACH:
+	case GUMSTIX_ARBOR43C:
 	case GUMSTIX_NO_EEPROM:
 	case GUMSTIX_EMPTY_EEPROM:
 		/* first lan chip */
-- 
2.1.4

