diff --git a/home/gumstix/yocto/build/tmp/work-shared/dragonboard-410c/kernel-source/arch/arm64/boot/dts/qcom/apq8016-sbc.dts b/home/gumstix/Documents/OV5640camera/WORKING20171228/qcom/apq8016-sbc.dts
index 825f489..19e96c6 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-sbc.dts
+++ b/arch/arm64/boot/dts/qcom/apq8016-sbc.dts
@@ -14,6 +14,7 @@
 /dts-v1/;
 
 #include "apq8016-sbc.dtsi"
+#include "dragonboardov5640.dtsi"
 
 / {
 	model = "Qualcomm Technologies, Inc. APQ 8016 SBC";
diff --git a/home/gumstix/yocto/build/tmp/work-shared/dragonboard-410c/kernel-source/arch/arm64/boot/dts/qcom/diff.patch b/home/gumstix/yocto/build/tmp/work-shared/dragonboard-410c/kernel-source/arch/arm64/boot/dts/qcom/diff.patch
deleted file mode 100644
index e69de29..0000000
diff --git a/home/gumstix/Documents/OV5640camera/WORKING20171228/qcom/dragonboardov5640.dtsi b/home/gumstix/Documents/OV5640camera/WORKING20171228/qcom/dragonboardov5640.dtsi
new file mode 100644
index 0000000..405799e
--- /dev/null
+++ b/arch/arm64/boot/dts/qcom/dragonboardov5640.dtsi
@@ -0,0 +1,51 @@
+
+&camss {
+	ports {
+		port@0 {
+			reg = <0>;
+			csiphy0_ep: endpoint {
+				status = "ok";
+				clock-lanes = <1>;
+				data-lanes = <0 2>;
+				qcom,settle-cnt = <0xe>;
+				remote-endpoint = <&ov5640_ep>;
+			};
+		};
+	};
+};
+
+&blsp_i2c6 {
+	status = "ok";
+
+	camera_rear@78 {
+		// This port isn't really connected to I2C-6
+		// This is a hack until CCI driver is a full i2c driver.
+		// The device connections listed are for port J3 - MIPI CSI0
+		status = "ok";
+		compatible = "ovti,ov5640";
+		reg = <0x78>;
+
+		ovti,vflip;
+		qcom,use_cci;
+		enable-gpios = <&msmgpio 36 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&msmgpio 12 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&camera_rear_default>;
+
+		clocks = <&gcc GCC_CAMSS_MCLK0_CLK>;
+		clock-names = "xclk";
+		clock-frequency = <23880000>;
+
+		vdddo-supply = <&camera_vdddo_1v8>;
+		vdda-supply = <&camera_vdda_2v8>;
+		vddd-supply = <&camera_vddd_1v5>;
+
+		port {
+			ov5640_ep: endpoint {
+				clock-lanes = <1>;
+				data-lanes = <0 2>;
+				remote-endpoint = <&csiphy0_ep>;
+			};
+		};
+	};
+};
