From f046c21c5a65f70519ceaa5e561c8d63bfbf6a6e Mon Sep 17 00:00:00 2001
From: Steve Sakoman <steve@sakoman.com>
Date: Sat, 10 Nov 2012 21:08:23 -0800
Subject: [PATCH 03/10] duovero: add back omap4_mux_init

---
 arch/arm/mach-omap2/board-duovero.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-omap2/board-duovero.c b/arch/arm/mach-omap2/board-duovero.c
index 4f53561..88b2f83 100644
--- a/arch/arm/mach-omap2/board-duovero.c
+++ b/arch/arm/mach-omap2/board-duovero.c
@@ -48,6 +48,7 @@
 #include "common.h"
 #include "control.h"
 #include "hsmmc.h"
+#include "mux.h"
 #include "common-board-devices.h"
 
 #define HDMI_GPIO_LS_OE		41
@@ -332,6 +333,9 @@ static struct platform_device *duovero_devices[] __initdata = {
 
 static void __init duovero_init(void)
 {
+	int package = OMAP_PACKAGE_CBS;
+
+	omap4_mux_init(NULL, NULL, package);
 	duovero_i2c_init();
 	platform_add_devices(duovero_devices, ARRAY_SIZE(duovero_devices));
 	omap_serial_init();
-- 
1.7.10.4

