From 29f8c28e3adeccb8572c6d2ecd2f93178770d8d8 Mon Sep 17 00:00:00 2001
From: HarryHu1 <harry.hu@gumstix.com>
Date: Wed, 19 Feb 2020 15:29:51 -0800
Subject: [PATCH] Add wlink8 and sound codec overlays to the default build.

---
 arch/arm/boot/dts/overlays/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index c078a2566..c269d4937 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -146,6 +146,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	sc16is752-spi1.dtbo \
 	sdhost.dtbo \
 	sdio.dtbo \
+	sdio-wlink8.dtbo \
 	sdtweak.dtbo \
 	sh1106-spi.dtbo \
 	smi.dtbo \
@@ -178,6 +179,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	tc358743.dtbo \
 	tc358743-audio.dtbo \
 	tinylcd35.dtbo \
+	tlv320aic3x-simple.dtbo \
 	tpm-slb9670.dtbo \
 	uart0.dtbo \
 	uart1.dtbo \
-- 
2.25.0

