From 09619b19c337ac7b2a28d816bfc9a83639bdc574 Mon Sep 17 00:00:00 2001
From: "Andrew C. Smith" <acsmith@gumstix.com>
Date: Wed, 3 Jul 2019 23:17:50 -0400
Subject: [PATCH] Add wlink8 and sound codec overlays to the default build.

---
 arch/arm/boot/dts/overlays/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index 10856e4..f5e35cc 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -131,6 +131,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	sc16is752-spi1.dtbo \
 	sdhost.dtbo \
 	sdio.dtbo \
+	sdio-wlink8.dtbo \
 	sdtweak.dtbo \
 	smi.dtbo \
 	smi-dev.dtbo \
@@ -160,6 +161,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	tc358743.dtbo \
 	tc358743-audio.dtbo \
 	tinylcd35.dtbo \
+	tlv320aic3x-simple.dtbo \
 	tpm-slb9670.dtbo \
 	uart0.dtbo \
 	uart1.dtbo \
-- 
2.7.4

